src/rules/jsxNoLambdaRule.ts

```diff
@@ -28,23 +29,34 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class JsxNoLambdaWalker extends Lint.RuleWalker {
+    private isInJsxAttribute = false;
     private isInJsxExpression = false;
 
-    public visitJsxExpression(node: ts.JsxExpression) {
+    protected visitNode(node: ts.Node) {
+        if (nodeIsKind<ts.JsxAttribute>(node, ts.SyntaxKind.JsxAttribute)) {
+            this.isInJsxAttribute = true;
+            super.visitNode(node);
+            this.isInJsxAttribute = false;
+        } else {
+            super.visitNode(node);
+        }
+    }
+
+    protected visitJsxExpression(node: ts.JsxExpression) {
         this.isInJsxExpression = true;
         super.visitJsxExpression(node);
         this.isInJsxExpression = false;
     }
 
-    public visitFunctionExpression(node: ts.FunctionExpression) {
-        if (this.isInJsxExpression) {
+    protected visitFunctionExpression(node: ts.FunctionExpression) {
+        if (this.isInJsxAttribute && this.isInJsxExpression) {
```

Comment:
No, it's `JsxElement` -> `JsxAttribute` -> `JsxExpression` because `JsxExpression` is just the `{code here}` part of things. But, lambdas inside a `JsxAttribute` would always be inside a `JsxExpression` as far as I can tell, so good call, we can probably eliminate the `isInJsxExpression` field and check